DeviceBridgeTests/Providers/StorageProviderTests.cs

```diff
@@ -1,40 +1,373 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Text.RegularExpressions;
 using System.Threading.Tasks;
+using DeviceBridge.Models;
+using DeviceBridge.Services;
+using Microsoft.QualityTools.Testing.Fakes;
+using Moq;
+using NLog;
 using NUnit.Framework;
 
 namespace DeviceBridge.Providers.Tests
 {
     [TestFixture]
     public class StorageProviderTests
     {
+        private Mock<IEncryptionService> _encriptionServiceMock;
+        private StorageProvider _storageProvider;
+
+        [SetUp]
+        public async Task Setup()
+        {
+            _encriptionServiceMock = new Mock<IEncryptionService>();
+            _encriptionServiceMock.Setup(p => p.Encrypt(It.IsAny<Logger>(), It.IsAny<string>())).Returns((Logger _, string s) => Task.FromResult(s));
+            _encriptionServiceMock.Setup(p => p.Decrypt(It.IsAny<Logger>(), It.IsAny<string>())).Returns((Logger _, string s) => Task.FromResult(s));
+            _storageProvider = new StorageProvider("", _encriptionServiceMock.Object);
+        }
+
+        [Test]
+        public async Task ListAllSubscriptionsOrderedByDeviceId()
```

Comment:
Test could be more explainable as description from method name, common pattern I saw in Microsoft, they used [Method Name]-Using[xxxx]-Expects/Throws[xxx].

Reply:
Since we have one test method per method of the interface, I've added description fields to all test cases t explain the scenario we're testing.

Reply:
Description look good to me as well.